### PR TITLE
github: Use cache when DB download fails

### PR DESCRIPTION
**Pull request by hamistao**
This updates the Trivy workflow to cache the vulnerability database when the download is successful and, when it is not, use the latest cached database instead. Motivated by some failures seen on some recent runs: https://github.com/canonical/microcloud/actions/runs/11638102909/job/32412374433

### CODE REVIEW

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -49,7 +66,7 @@ jobs:
           ref: refs/heads/main
 
   trivy-snap:
-    name: Trivy vulnerability scanner - Snap
+    name: Trivy - Snap
```
Comment (roosterfish) [inline .github/workflows/security.yml:69]: Unrelated to this PR, but can we run the check for multiple snaps channels using a matrix build? 
I am thinking we could check `2/candidate` and `2/stable`. 

But we would probably need an option to group the results so they are not getting mixed up.
I have quickly checked the action's [docs](https://github.com/github/codeql-action/blob/v3/upload-sarif/action.yml) and it looks there is a `category` setting we could potentially use. 
Have you maybe tried something like this already?

Comment (hamistao) [reply .github/workflows/security.yml:69]: I did look into this `category` setting but it doesn't help filter the scan results on the security tab :/

What we do on LXD is we place results for each `{version}/stable` channel in its corresponding branch. So `5.21/stable` scan results are tagges as belonging to `stable-5.21` branch.

Comment (hamistao) [reply .github/workflows/security.yml:69]: @roosterfish  What do you think of placing `2/stable` results in `v2-edge` and `3/stable` in the main branch?

Comment (roosterfish) [reply .github/workflows/security.yml:69]: Sounds like a good plan.

Comment (roosterfish) [reply .github/workflows/security.yml:69]: Ah but we don't yet have `3/stable`. 
There will be a `2/stable` starting from next week we can check.
And going forward there will be only be a `3/edge` but I assume we can use this channel instead.

Comment (roosterfish) [reply .github/workflows/security.yml:69]: Maybe for now we can use `3/edge` in the `main` branch and `2/candidate` in the `v2-edge` until we release the LTS.

**Review by roosterfish — APPROVED** [review]

**Comment by roosterfish** [comment]
Thanks!

**Pull request merged**